spec/datadog/core/utils/safe_dup_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'datadog/core/utils/safe_dup'
+
+RSpec.describe Datadog::Core::Utils::SafeDup do
+  describe '.frozen_or_dup' do
+    context 'when given a frozen string' do
+      it 'returns the original input' do
+        input = 'a_frozen_string'.freeze
+
+        result = described_class.frozen_or_dup(input)
+
+        expect(result).to eq(input)
+        expect(result).to equal(input)
```

Comment:
Minor: This can be simplified a bit, because we only need to check `#equal?`, as that's the one that tests the object references and should never be overridden.

Rspec has a nice alias for equal that I like to use in these situations:

```suggestion
        expect(result).to be(input)
```